### PR TITLE
feat: add client log shipping from hmi client to hmi server

### PR DESCRIPTION
**Pull request by tangopapatime**
# Description

First part of the setting up a logging framework to capture user logs in a structured manner.  This will allow to measure user interactions and issues that occurred during usage.

Currently uses `vue-logger-plugin` and it's `afterHook` to buffer log messages and then post the messages periodically every 5 seconds.

Resolves #(issue)
Resolves the first part of setting up the instrumentation to

ship client calls to the server
the server saves structured logs

https://github.com/DARPA-ASKEM/TERArium/issues/462#event-8213619647

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -11377,6 +11378,13 @@ send@latest:
   languageName: node
   linkType: hard
 
+"vue-logger-plugin@npm:2.2.3":
```
Comment (tangopapatime) [inline yarn.lock:11381]: Definitely don't really know how/what exactly needs to be checked in to get a new package working.

Comment (YohannParis) [reply yarn.lock:11381]: What do you mean?

Comment (tangopapatime) [reply yarn.lock:11381]: Just meant that I'm not sure I added the package in all the places it needs to be.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	@Inject
+	SecurityIdentity securityIdentity;
+
+	@Inject
+	LoggingService loggingService;
+
+	@POST
+	@Path("/logs")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java:31]: The default produce/consume type is APPLICATION_JSON, so you won't need these here (especially with the Produce/Consume at the top of the class too)

File: yarn.lock (hunk 1 above)
Comment (YohannParis) [reply yarn.lock:11381]: The right way to add packages is:
- go within `TERArium/packages/client/hmi-client`
- install package with `yarn add --exact [package_name]`
- make sure `yarn.lock` is part of the commit

Then you are all good, and it looks like this is what you did.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	public Response letsDoThis(LoggingService logData) {
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java:32]: Also, may need a more description function name ;)

**Review by dvince2 — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 1 above)
Comment (tangopapatime) [reply yarn.lock:11381]: gotcha.  Thanks @YohannParis .

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java (hunk 2 above)
Comment (tangopapatime) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java:31]: Definitely didn't know this as Quarkus virgin :P

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java (hunk 3 above)
Comment (tangopapatime) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/loggingservice/LoggingResource.java:32]: lol, oops, yeah forgot.  glad I didn't use my usual temp naming of using curse words.

**Review by dvince2 — DISMISSED** [review]

File: packages/client/hmi-client/src/main.ts [hunk 4]
```diff
@@ -10,6 +11,7 @@ const app = createApp(App);
 app.use(createPinia());
 app.use(router);
 app.use(PrimeVue, { ripple: true });
+app.use(createLogger({}));
```
Comment (mwdchang) [inline packages/client/hmi-client/src/main.ts:14]: This looks a little strange, is the usage correct? I would have expected that createLogger to be called only once but we are calling it twice?

We have
```
import { createLogger } from 'vue-logger-plugin';
app.use(createLogger({}));
```

shouldn't this be 

```
import { logger } from '@/utils/logger-something-something';
app.use(logger);
```

where the logger instance is created in the util?

File: packages/client/hmi-client/src/utils/logger.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		if (!this.isEmpty()) {
+			await API.post(`/logs/`, {
+				logs: this.getLogBuffer()
+			})
+				.then(() => {
+					this.clearLogs();
+				})
+				.catch((error) => {
+					console.error(error);
+				});
```
Comment (mwdchang) [inline packages/client/hmi-client/src/utils/logger.ts:55]: Would prefer if we can stick with either async/await-style, or promise-style.  But not both at the same time as it makes it a bit harder to understand.

I think the code here, as-is,  it technically works but the await is redundant.

File: packages/client/hmi-client/src/main.ts (hunk 4 above)
Comment (tangopapatime) [reply packages/client/hmi-client/src/main.ts:14]: no you're right.  found it strange that this worked in the first place.

**Review by dvince2 — APPROVED** [review]
lgtm

File: packages/client/hmi-client/src/utils/logger.ts (hunk 5 above)
Comment (tangopapatime) [reply packages/client/hmi-client/src/utils/logger.ts:50]: fixed.

**Pull request merged**